inc/deprecated/deprecated.php

```diff
@@ -50,7 +50,6 @@ function rocket_sanitize_cookie( $cookie ) {
 	 */
 	function set_rocket_cloudflare_async( $cf_rocket_loader ) {
 		_deprecated_function( __FUNCTION__, '2.8.16', 'set_rocket_cloudflare_rocket_loader()' );
-		return set_rocket_cloudflare_rocket_loader( $cf_rocket_loader );
```

Comment:
Why remove it?

Reply:
> You should remove these lines - @Tabrisrp commented on `deprecated.php` file.

Reply:
OK thanks Cristina. Then, @Tabrisrp, why remove it? ^^

Reply:
When I did the review it was showing the line twice, now it's not anymore. Git doing git things  I guess.

So yeah in fact we need to keep it, sorry about that

Reply:
There are 2 functions in there: `set_rocket_cloudflare_async()` and `set_rocket_cloudflare_cache_lvl()`. Should those call the new functions from CloudflareSubscriber.php?
`Cloudflare_Subscriber::set_cache_level()` 
`Cloudflare_Subscriber::set_rocket_loader()`

Also for all deprecated functions, it will throw an error because `$GLOBALS['rocket_cloudflare']` is not set anymore. How should this be handled?
https://github.com/wp-media/wp-rocket/blob/7723e789d158d2bf00ee5da49ef9158061e158eb/inc/functions/cloudflare.php#L195

Reply:
This is something we never discussed actually, or I'm not remembering it. So that might be the good opportunity :)

I see 3 ways to handle the function behavior:
- keep the function as it was: its behavior doesn't change. We need to be careful with the functions it uses internally though, they may change in the future.
- replace its content by the new function it is replaced by (this sentence doesn't make sense ^^): its behavior may not change at first, but the new function may change in the future, maybe by adding a parameter to it for example, or simply by changing its internal behavior. If this happens, things will break if the deprecated function is still used.
- empty the function, so it does nothing anymore: things that were working with this function will not anymore, but will not "break".

Personally I prefer the first case, it feels safer imho.

About `$GLOBALS['rocket_cloudflare']`, and global vars in general. This one seems tricky to me. Since global vars are public, we shouldn't remove them in theory. On the other hand, keeping them will result in useless process. I did a quick search but couldn't find a way to lazyload it.

Maybe we could add something like that at the beginning of the functions that use the global var:
```php
if ( ! isset( $GLOBALS['rocket_cloudflare'] ) ) {
    $GLOBALS['rocket_cloudflare'] = get_rocket_cloudflare_instance();
}
```

What do you think about all this?

Reply:
For the `$GLOBALS['rocket_cloudflare'] = get_rocket_cloudflare_instance();` I would suggest to totally remove it because this was the cause of too many hits to Cloudflare API. So this GLOBAL should not be used at all from my point of view.
But at this point, the deprecated methods will show an error because of the missing GLOBAL so I think we should discuss the approach for deprecations.

Reply:
OK, thanks, I was not aware of this. Indeed, we should discuss all that next week.

Reply:
I don't think any of our cloudflare functions were used externally by other plugins or anything, so the deprecation is there to prevent any error after the update mainly.

My opinion is to remove any call to the globals, and make the functions who used it to do nothing, only make sure they don't trigger any error.

Reply:
So, if I remove the global from each deprecated function, then I should remove the full code inside the function because it will throw an notice with undefined:
https://github.com/wp-media/wp-rocket/blob/38e9239a402d5064b52c4e63af76e95532fb2f0d/inc/deprecated/3.5.php#L244-L254

So the solution in this case is to keep only the `_deprecated_function` in action for each deprecated cloudflare function?

Reply:
For me that's fine yes

Reply:
Removed all function content for all deprecated cloudflare functions which contained `$GLOBALS['rocket_cloudflare']; `